stdlib/private/OSLog/OSLogMessage.swift

```diff
@@ -0,0 +1,309 @@
+  public var arguments: OSLogArguments
+
+  /// The first summary byte, as defined by the OS logging system, that
+  /// summarizes the nature and type of the arguments.
+  public var preamble: UInt8
+
+  /// The second summary byte, which corresponds to the number of arguments i.e,
+  /// number of interpolated expressions. This will be determined on the fly in
+  /// order to support concatenation and interpolation of instances of
+  /// `OSLogMessage`.
+  public var argumentCount: UInt8
+
+  public init(literalCapacity: Int, interpolationCount: Int) {
+    // TODO: format string must be fully constructed at compile time.
+    // The parameters `literalCapacity` and `interpolationCount` are ignored.
+    formatString = ""
+    arguments = OSLogArguments()
+    preamble = 0
+    argumentCount = 0
+  }
+
+  public mutating func appendLiteral(_ literal: String) {
+    formatString += literal.percentEscapedString
+  }
+
+  /// Define interpolation for expressions of type Int. This definition enables
+  /// passing a formatting option and a privacy qualifier along with the
+  /// interpolated expression as shown below:
+  ///
+  ///         "\(x, format: .hex, privacy: .private\)"
+  ///
+  /// - Parameters:
+  ///  - number: the interpolated expression of type Int, which is autoclosured.
+  ///  - format: a formatting option available for Int types, defined by the
+  ///    enum `IntFormat`.
+  ///  - privacy: a privacy qualifier which is either private or public.
+  ///    The default is public.
+  public mutating func appendInterpolation(
+    _ number: @autoclosure @escaping () -> Int,
+    format: IntFormat = .decimal,
+    privacy: Privacy = .public
+  ) {
+    guard argumentCount < maxOSLogArgumentCount else { return }
+
+    addIntHeadersAndFormatSpecifier(
+      format,
+      isPrivate: privacy == .private,
+      bitWidth: Int.bitWidth,
+      isSigned: true)
+    arguments.append(number)
+  }
+
+  /// Construct/update format string and headers from the qualifiers (of the
+  /// interpolated expression) passed as parameters.
+  ///
+  /// All arguments to this function must be known at compile time.
+  public mutating func addIntHeadersAndFormatSpecifier(
+    _ format: IntFormat,
+    isPrivate: Bool,
+    bitWidth: Int,
+    isSigned: Bool
+  ) {
+    formatString += getIntegerFormatSpecifier(
+      format,
+      isPrivate: isPrivate,
+      bitWidth: bitWidth,
+      isSigned: isSigned)
+
+    addArgumentHeaders(
+      // Always explicitly set either public or private flag bit.
+      flagAndType: isPrivate ? 1 : 2,
+      byteSize: UInt8(bitWidth / 8))
+
+    updateSummaryBytes(isPrivate: isPrivate)
+  }
+
+  // Update the summary bytes: preamble and argument count.
+  public mutating func updateSummaryBytes(isPrivate: Bool) {
+    preamble |= isPrivate ? 1 : 0
+    argumentCount += 1
+  }
+
+  // Append the given argument header and size.
+  public mutating func addArgumentHeaders(flagAndType: UInt8, byteSize: UInt8) {
```

Comment:
is "byteSize" the size of the argument? If so, then perhaps "sizeInBytes" or just "size" along with a doc comment describing the parameter?